src/compile/model.ts

```diff
@@ -241,7 +241,7 @@ export abstract class Model {
   public abstract assembleLayoutSignals(): VgSignal[];
 
   public assembleScales(): VgScale[] {
-    return assembleScale(this);
+    return assembleScalesForModel(this);
   }
```

Comment:
As far as I can tell, `Model` and every `Model` subclass has an identical `assembleScales` function? My vote would be to remove them from the subclasses to reduce the number of codepaths.